### ISSUE TITLE
API: querying versions with filter all_without_unlisted fails with a permission error if there are no listed versions

### ISSUE THREAD

**Issue by kewisch** [mozilla/addons-server#11187]
### Describe the problem and steps to reproduce it:
I'm trying to access https://reviewers.addons.mozilla.org/api/v4/addons/addon/%7B132cb2fd-a6ae-45d2-84cf-b48d591f037d%7D/versions/?filter=all_without_unlisted . I'm including necessary Authorization headers which seem to work.

The specific add-on does not have any listed versions, which is expected.

### What happened?
* Sending the request, I get an error `You do not have permission to perform this action`.
* It works fine if I use `?filter=all_with_unlisted`

### What did you expect to happen?
* The request should return zero results, but not bail with a permission error

**Comment by EnTeQuAk** [comment]
Some very short investigation shows that the requested version does not have any listed versions so if you don't have the `ReviewTools:View` permission you seem to be getting a 403. Haven't looked at this more.

**Comment by diox** [comment]
`ReviewTools:View` permission shouldn't be needed for this:
- `AddonVersionViewSet` sets `self.permission_classes = [AnyOf(AllowReviewerUnlisted, AllowAddonAuthor)]`
- It then calls `check_object_permissions()` using the addon object
- `AllowReviewerUnlisted` does a call to `check_unlisted_addons_reviewer()`  as long as `obj.has_unlisted_versions() or not obj.has_listed_versions()` is `True` (which should be the case here)
- Which does a action_allowed check with `ADDONS_REVIEW_UNLISTED`
- Which should be `Addons:ReviewUnlisted`

Not sure what's going on exactly, I'll investigate when I get back.

**Comment by diox** [comment]
Err, after typing a big comment before I noticed that you used `all_without_unlisted` and not `all_with_unlisted`. I blame lack of sleep. 

If there are only unlisted version, and you are using `all_without_unlisted`, then yeah I can see why we'd fail - the whole `ReviewTools:View` is kinda misleading though - it's just something added for backwards-compatibility with something similar that was in place before. The real reason behind this failure is simply that it's not a use case we had really considered when making the permission classes.

**Comment by eviljeff** [comment]
@kewisch what permissions do you have exactly on prod?  I'm having trouble reproducing this in a test.  It can't just `ReviewerTools:View` because you wouldn't be able to do `all_with_unlisted` successfully and you said that worked.  (`ReviewTools:View` doesn't exist)

**Comment by jvillalobos** [comment]
He's not a full admin, but he's in the Staff group (Addons:Configure,Addons:ContentReview,Addons:Edit,Addons:PostReview,Addons:Review,Addons:ReviewUnlisted,Addons:ViewDeleted,Admin:Tools,Collections:Edit,CollectionStats:View,Personas:Review,ReviewerAdmin:Tools,Reviews:Edit,Stats:View,Users:Edit,Addons:ThemeReview,Ratings:Moderate) and the Reviewers: Admin group (Reviews:Admin)

**Comment by ioanarusiczki** [comment]
@eviljeff  I tested this on AMO dev, I added the user into the `Staff` and `Reviewers:Admin` Group. 
For https://reviewers.addons-dev.allizom.org/api/v4/addons/addon/503036/versions/?filter=all_without_unlisted when the addon has only unlisted versions I receive the following response in Postman:

![api resonse for results](https://user-images.githubusercontent.com/33448286/57440664-17e03d80-7251-11e9-9ec2-28cc7002677f.png)

I did not receive permission errors. 

One question: I don't know if the `results` parameter should return `[ ]`. If it should be `null` I could file a followup.
